# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ShareableCanvasRenderer.cpp
# Commit: c09a51622e98
# Full Hash: c09a51622e988e81077c75e3aaa5a49df66284b3
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-22 16:44:08
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 737807563dd5 -r c09a51622e98 gfx/layers/ShareableCanvasRenderer.cpp
--- a/gfx/layers/ShareableCanvasRenderer.cpp	Thu Mar 21 23:14:26 2019 +0000
+++ b/gfx/layers/ShareableCanvasRenderer.cpp	Fri Mar 22 03:15:14 2019 +0000
@@ -177,7 +177,8 @@
   return CanvasClient::CanvasClientSurface;
 }
 
-void ShareableCanvasRenderer::UpdateCompositableClient() {
+void ShareableCanvasRenderer::UpdateCompositableClient(
+    wr::RenderRoot aRenderRoot) {
   if (!CreateCompositable()) {
     return;
   }
@@ -197,14 +198,16 @@
       gfxCriticalNote << "BufferProvider::SetForwarder failed";
       return;
     }
-    mCanvasClient->UpdateFromTexture(mBufferProvider->GetTextureClient());
+    mCanvasClient->UpdateFromTexture(mBufferProvider->GetTextureClient(),
+                                     aRenderRoot);
   } else {
-    mCanvasClient->Update(gfx::IntSize(mSize.width, mSize.height), this);
+    mCanvasClient->Update(gfx::IntSize(mSize.width, mSize.height), this,
+                          aRenderRoot);
   }
 
   FireDidTransactionCallback();
 
-  mCanvasClient->Updated();
+  mCanvasClient->Updated(aRenderRoot);
 }
 
 }  // namespace layers